step-capstone/src/components/Trip.js

```diff
@@ -112,78 +21,59 @@ export default class Trip extends React.Component {
         this.handleAddItem = this.handleAddItem.bind(this);
     }
 
-    fetchData() {
-        // TODO: fetch from datastore
-        return testData;
-    }
-
     componentDidMount() {
-        let data = this.fetchData();
-        this.setState({
-            items: data
-        })
+      let travelObjectList = [];
+      this.context.getData(this.state.reference)
+      .then(data => {
+        data.data().travelObjects.forEach(travelObject => {
+          travelObjectList.push(travelObject)
+        });
+        this.setState({items : travelObjectList});
+      })
+      .catch(error => {console.log("Error Getting Trip Data")});
```

Comment:
If you're logging an error, make sure to include the original error message as well or it will be hard to debug. Let me know if react does this automatically but I couldn't find documentation of that. This applies to pretty much all your error handling.

Reply:
done, thanks